src/script/api/script_vehicle.cpp

```diff
@@ -75,7 +75,7 @@
 {
 	EnforceCompanyModeValid(VEHICLE_INVALID);
 	EnforcePrecondition(VEHICLE_INVALID, ScriptEngine::IsBuildable(engine_id));
-	EnforcePrecondition(VEHICLE_INVALID, cargo == CT_INVALID || ScriptCargo::IsValidCargo(cargo));
+	EnforcePrecondition(VEHICLE_INVALID, !IsValidCargoID(cargo) || ScriptCargo::IsValidCargo(cargo));
```

Comment:
Same as my comment above, although the other one is documentation so it's probably more critical for it to make sense to humans.